### DISCUSSION TITLE
Leaking information to Google

### DISCUSSION THREAD

**Discussion by OctarineLobster** [usememos/memos#3651] [Q&A]
for links in a memo, it appears to reach out to google for link metadata
I don't see any controls on the settings pages for this behaviour, can this be disabled? I can raise a feature request if there's no way to turn this off.

Note: I would not consider this "privacy-first" behaviour

![Memo](https://github.com/usememos/memos/assets/31774832/741d217a-a0a3-41a8-9c1b-df3024ca8ba5)

**Comment by andrigamerita** [comment]
There are some options to tackle this issue:
* Use the server as a proxy to load the images, so the server can contact Google instead of the browser (would need to be implemented in the server, obviously)
* Have a setting for the client for which favicon provider to use (the same service is offered by others, such as DuckDuckGo).
* The client should render the favicon images with attribute `loading="lazy"`, so that it would be sufficient to add the following CSS to block the browser from requesting the images altogether: `body > div.base-Popper-root[data-popper-placement] > div > div > img[src^="https://www.google.com/s2/favicons?"] { display: none; visibility: hidden; }`

Maybe you should indeed convert this discussion to a feature request issue, because these are all things that must be worked on inside the code, there is no other proper way to do it.

But, in the meantime actually, if you have Memos set up behind a reverse proxy (nginx, apache, etc...), you can modify the client-side code yourself to fix this, without having to compile from source.
You can see in the network tab of your devtools which JS file starts the request to Google (`index-....js`)... so, you can download it as a text file, modify it in the following way, and configure your reverse proxy so that it serves your modified copy instead of the original file.
Look inside the file for the piece of text starting with `https://www.google.com/s2/favicons`, and either modify it to contact another favicon service instead of Google's, or if you don't care about the feature just make the function return some empty text (basically replace the whole `{return 'https://...'}` with `{return '';}`).

**Reply by MehadND** [reply]
I was one of the main contributors to this feature, so please do share with me if you were able to resolve this issue or not?

**Comment by Novapixel1010** [comment]
bumping this. Would be great to just disable it entirely in the system settings for memos.

**Comment by Novapixel1010** [comment]
I did notice in v0.25.0 that there is a feature called `Enable link preview` which I have toggled off.  That does fix the privacy issue. Here is the network log:

<img width="1687" height="687" alt="memos-1" src="https://github.com/user-attachments/assets/8f0f1334-99d9-4485-a823-66cd3aac8359" />

I also noticed when you have it enabled it doesn't look anything up until you hover over the link. I have not checked how the it is handle in mobile. With that said toggling off `Enable link preview` fixes the privacy concern.
